diff -uNr gnome-documents-3.34.0.orig/meson.build gnome-documents-3.34.0/meson.build
--- gnome-documents-3.34.0.orig/meson.build	2022-07-31 18:31:03.494645942 -0300
+++ gnome-documents-3.34.0/meson.build	2022-07-31 18:52:10.949647973 -0300
@@ -68,6 +68,7 @@
   dependency('gobject-introspection-1.0', version: '>= 1.31.6'),
   dependency('gtk+-3.0', version: '>= 3.22.15'),
   dependency('libsoup-2.4', version: '>= 2.41.3'),
+  dependency('gdlib', version: '>= 2.3.3'),
   dependency('tracker-control-2.0', version: tracker_req_version),
   dependency('tracker-sparql-2.0', version: tracker_req_version),
   dependency('webkit2gtk-4.0', version: '>= 2.6.0'),
