From: Kalev Lember <klember@redhat.com>
Date: Wed, 26 Feb 2020 14:11:38 +0100
Subject: getting-started: Fix the build with inkscape 1.0

Bug-RedHat: https://bugzilla.redhat.com/show_bug.cgi?id=1799427
Forwarded: https://gitlab.gnome.org/GNOME/gnome-documents/-/merge_requests/31
---
 getting-started/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/getting-started/meson.build b/getting-started/meson.build
index 4cb4fe7..9d9c63d 100644
--- a/getting-started/meson.build
+++ b/getting-started/meson.build
@@ -21,7 +21,7 @@ foreach lingua: linguas
       source,
       input: join_paths(lingua, source + '.svg'),
       output: source + '.pdf',
-      command: [inkscape, '-z', '-A', '@OUTPUT@', '@INPUT@']
+      command: [inkscape, '--export-filename', '@OUTPUT@', '@INPUT@']
     )
   endforeach
 
